### PR TITLE
Add docs about --server

### PR DESCRIPTION
**Pull request by bmw**

### CODE REVIEW

**Comment by bmw** [comment]
We can update this to include LE's ACMEv2 URL when it is public.

File: docs/using.rst [hunk 1]
```diff
@@ -801,6 +801,27 @@ Example usage for DNS-01 (Cloudflare API v4) (for example purposes only, do not
...
+
+By default, Certbot uses Let's Encrypt's initial production server at
+https://acme-v01.api.letsencrypt.org/. You can tell Certbot to use a
+different CA by providing ``--server`` on the command line or in a
+:ref:`configuration file <config-file>` with the URL of the server's
+ACME directory. For example, to test against Let's Encrypt's ACMEv2
+staging server, you would add ``--server
+https://acme-staging-v02.api.letsencrypt.org/directory`` to the command
+line. Certbot will automatically change the version of the ACME protocol
+used based on the contents served at the provided URL.
```
Comment (ohemorange) [inline docs/using.rst]: Certbot will automatically ~change the~**select which** version of the ACME protocol to use based on the contents served at the provided URL.

File: docs/using.rst [hunk 2]
```diff
@@ -801,6 +801,27 @@ Example usage for DNS-01 (Cloudflare API v4) (for example purposes only, do not
...
+
+If you use ``--server`` to specify an ACME CA that implements a newer
+version of the spec, you may want to obtain a certificate for a wildcard
```
Comment (ohemorange) [inline docs/using.rst]: you may want to --> you will be able to

Comment (bmw) [reply docs/using.rst]: I used "you may be able to" because it depends on the CA's policy.

**Comment by bmw** [comment]
I added a couple new changes such as documenting `--server` when talking about running the DNS plugins in Docker and using the production ACMEv2 as the example rather than the staging one.

**Comment by bmw** [comment]
@ohemorange, I'm not blocking the initial website update on this, but if you could prioritize reviewing this when you get a chance, I'd appreciate it as not setting the `--server` flag has been a source of problems on the community forum.

File: docs/using.rst (hunk 2 above)
Comment (ohemorange) [reply docs/using.rst]: cool

Comment (ohemorange) [reply docs/using.rst]: cool

**Pull request merged**